src/algebra/order/floor.lean

```diff
@@ -81,6 +81,12 @@ def ceil : α → ℕ := floor_semiring.ceil
 notation `⌊` a `⌋₊` := nat.floor a
 notation `⌈` a `⌉₊` := nat.ceil a
 
+lemma exists_nat_ge (x : α) : ∃ (n : ℕ), x ≤ ↑n :=
+⟨⌈x⌉₊, (floor_semiring.gc_ceil x ⌈x⌉₊).mp rfl.le⟩
```

Comment:
```suggestion
⟨⌈x⌉₊, le_ceil _⟩
```
where `le_ceil` is further down the file.

Reply:
So the problem with this is that it would require picking apart which lemmas can be gotten with just an `ordered_semiring` and which require a `linear_ordered_semiring`. `le_ceil` in particular is one that only requires an `ordered_semiring`, but it's a bit tedious to pull them apart.

Reply:
If you'd like me to make an attempt I can, but it will be a significant refactoring of the file! (Basically, anything that uses `lt_iff_lt_of_le_iff_le` requires a linear order.)

Reply:
I think either:

1. You should put these lemmas in the `linear_ordered_semiring` section for now, if that doesn't actually break anything
2. You should make a new PR that does the tedious work (and only the tedious work).

I assume that 1 isn't an option. It doesn't really make sense to put off 2, as otherwise we're in the situation where we have an entire API we can't use in the place where you need it.